src/perform_step/linear_perform_step.jl

```diff
@@ -28,6 +28,42 @@ function perform_step!(integrator, cache::MagnusMidpointCache, repeat_step=false
   integrator.destats.nf += 1
 end
 
+function initialize!(integrator, cache::MagnusGauss4Cache)
+  integrator.kshortsize = 2
+  integrator.fsalfirst = cache.fsalfirst
+  integrator.fsallast = cache.k
+  resize!(integrator.k, integrator.kshortsize)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+  integrator.f(integrator.fsalfirst, integrator.uprev, integrator.p, integrator.t) # For the interpolation, needs k at the updated point
+  integrator.destats.nf += 1
+end
+
+function perform_step!(integrator, cache::MagnusGauss4Cache, repeat_step=false)
+  @unpack t,dt,uprev,u,p,alg = integrator
+  @unpack W,k,tmp = cache
+  mass_matrix = integrator.f.mass_matrix
+
+  L1 = integrator.f.f
+  update_coefficients!(L1,u,p,t+dt*(1/2+sqrt(3)/6))
+  L2 = integrator.f.f
+  update_coefficients!(L2,u,p,t+dt*(1/2-sqrt(3)/6))
+  A = Matrix(L1)
+  B = Matrix(L2)
+  # println("Here a $A")
+  # println("Here b $B")
+
+  if integrator.alg.krylov
+    u .= expv(dt,(A+B) ./ 2 + (dt*sqrt(3)) .* (B*A-A*B) ./ 12, u; m=min(alg.m, size(L1,1)), opnorm=integrator.opts.internalopnorm, iop=alg.iop)
```

Comment:
I think we want to do successive application, though doing the commutator will be the issue...